pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1381,9 +1381,7 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 		domain.Spec.MemoryBacking = &api.MemoryBacking{
 			HugePages: &api.HugePages{},
 		}
-		if val := vmi.Annotations[v1.MemfdMemoryBackend]; val != "false" {
```

Comment:
Please remove this deprecation from this PR as it does not have relation to the content.

Reply:
> as it does not have relation to the content.

I understand your point, but not sure I entirely agree in this specific context.
Currently, memory backing is being used in [certain scenarios](https://github.com/kubevirt/kubevirt/blob/v1.0.0-alpha.0/pkg/virt-launcher/virtwrap/converter/converter.go#L1379) (e.g. when huge pages / virtiofs is being used) and is **always** backed by `memfd`. When this code was merged, almost [3 years ago](https://github.com/kubevirt/kubevirt/pull/4027), there was a concern regarding `memfd`, therefore the annotation is being introduced.

As written in the PR itself [here](https://github.com/kubevirt/kubevirt/pull/4027#issuecomment-688650428):
> it makes sense [to add this annotation] if there are any concerns. We can always remove the check once it becomes more stable.

After 2.5~ years I think it's perfectly fine to remove it.

In addition, even today, the annotation is effectively broken, as it only affects huge pages and not virtiofs usage, as can be seen [here](https://github.com/kubevirt/kubevirt/blob/v1.0.0-alpha.0/pkg/virt-launcher/virtwrap/converter/converter.go#L1396).

So I definitely think it should be deprecated, and my opinion is that this PR is a good opportunity for a bit of refactoring. Since we're messing with memory backing here, I think it is relevant, but if you insist we can deprecate it in a follow-up PR.